So I have fix it but now I've a new error:

```
Traceback (most recent call last):
  File "/usr/lib/python3.4/asyncio/tasks.py", line 237, in _step
    result = next(coro)
  File "/home/test/home-assistant/homeassistant/core.py", line 292, in async_stop
    yield from self.loop.run_in_executor(None, self.pool.block_till_done)
  File "/usr/lib/python3.4/asyncio/base_events.py", line 431, in run_in_executor
    return futures.wrap_future(executor.submit(callback, *args), loop=self)
  File "/usr/lib/python3.4/concurrent/futures/thread.py", line 99, in submit
    raise RuntimeError('cannot schedule new futures after shutdown')
RuntimeError: cannot schedule new futures after shutdown
```

But with CTR-C work fine...

--- a/homeassistant/core.py
+++ b/homeassistant/core.py
@@ -190,10 +190,10 @@ class HomeAssistant(object):
             _LOGGER.info("Starting Home Assistant core loop")
             self.loop.run_forever()
         except KeyboardInterrupt:
-            pass
-        finally:
             self.loop.call_soon(stop_homeassistant)
             self.loop.run_forever()
+        finally:
+            self.loop.close()
 
     @asyncio.coroutine
     def async_start(self):

--- a/tests/test_core.py
+++ b/tests/test_core.py
@@ -56,11 +56,11 @@ class TestHomeAssistant(unittest.TestCase):
         self.hass.bus.listen_once(EVENT_HOMEASSISTANT_STOP,
                                   lambda event: calls.append(1))
 
-        #os.kill(os.getpid(), signal.SIGTERM)
+        os.kill(os.getpid(), signal.SIGTERM)
 
-        #self.hass.block_till_done()
+        self.hass.block_till_done()
 
-        #self.assertEqual(1, len(calls))
+        self.assertEqual(1, len(calls))
 
 
 class TestEvent(unittest.TestCase):

--- a/tests/test_core.py
+++ b/tests/test_core.py
@@ -56,11 +56,11 @@ class TestHomeAssistant(unittest.TestCase):
         self.hass.bus.listen_once(EVENT_HOMEASSISTANT_STOP,
                                   lambda event: calls.append(1))
 
-        os.kill(os.getpid(), signal.SIGTERM)
+        #os.kill(os.getpid(), signal.SIGTERM)
 
-        self.hass.block_till_done()
+        #self.hass.block_till_done()
 
-        self.assertEqual(1, len(calls))
+        #self.assertEqual(1, len(calls))
 
 
 class TestEvent(unittest.TestCase):

